x-pack/plugins/spaces/server/lib/request_inteceptors/on_post_auth_interceptor.test.ts

```diff
@@ -0,0 +1,379 @@
+import sinon from 'sinon';
+
+import { SavedObject } from 'src/server/saved_objects';
+import { Feature } from 'x-pack/plugins/xpack_main/types';
+import { convertSavedObjectToSpace } from '../../routes/lib';
+import { initSpacesOnPostAuthRequestInterceptor } from './on_post_auth_interceptor';
+import { initSpacesOnRequestInterceptor } from './on_request_interceptor';
+
+describe('onPostAuthRequestInterceptor', () => {
+  const sandbox = sinon.sandbox.create();
+  const teardowns: Array<() => void> = [];
+  const headers = {
+    authorization: 'foo',
+  };
+  let server: any;
+  let request: any;
+
+  const serverBasePath = '/';
+  const defaultRoute = '/app/custom-app';
+
+  beforeEach(() => {
+    teardowns.push(() => sandbox.restore());
+    request = async (
+      path: string,
+      spaces: SavedObject[],
+      setupFn: (server: Server) => null = () => null
+    ) => {
+      server = new Server();
+
+      interface Config {
+        [key: string]: any;
+      }
+      const config: Config = {
+        'server.basePath': serverBasePath,
+        'server.defaultRoute': defaultRoute,
+      };
+
+      server.decorate(
+        'server',
+        'config',
+        jest.fn(() => {
+          return {
+            get: jest.fn(key => {
+              return config[key];
+            }),
+          };
+        })
+      );
+
+      server.savedObjects = {
+        SavedObjectsClient: {
+          errors: {
+            isNotFoundError: (e: Error) => e.message === 'space not found',
+          },
+        },
+        getSavedObjectsRepository: jest.fn().mockImplementation(() => {
+          return {
+            get: (type: string, id: string) => {
+              if (type === 'space') {
+                const space = spaces.find(s => s.id === id);
+                if (space) {
+                  return space;
+                }
+                throw new Error('space not found');
+              }
+            },
+            create: () => null,
+          };
+        }),
+      };
+
+      server.plugins = {
+        spaces: {
+          spacesClient: {
+            getScopedClient: jest.fn(),
+          },
+        },
+        xpack_main: {
+          getFeatures: () =>
+            [
+              {
+                id: 'feature-1',
+                name: 'feature 1',
+                app: ['app-1'],
+              },
+              {
+                id: 'feature-2',
+                name: 'feature 2',
+                app: ['app-2'],
+              },
+              {
+                id: 'feature-4',
+                name: 'feature 4',
+                app: ['app-1', 'app-4'],
+              },
+            ] as Feature[],
+        },
+      };
+
+      let basePath: string | undefined;
+      server.decorate('request', 'getBasePath', () => basePath);
+      server.decorate('request', 'setBasePath', (newPath: string) => {
+        basePath = newPath;
+      });
+
+      initSpacesOnRequestInterceptor(server);
```

Comment:
Would you mind explaining why we have to `initSpacesOnRequestInterceptor` as well?

Reply:
Yep, the `onPostAuth` interceptor depends on the `onRequest` interceptor running first, as it's responsible for parsing out the space id and rewriting the URL. I was torn between duplicating that logic here vs just including the already tested `onRequest` inceptor, and opted for the latter

Reply:
That makes sense, you mind adding a comment specifying why we're doing so?